### ISSUE TITLE
Support for footnotes

### ISSUE THREAD

**Issue by mateusfccp** [lvjr/tabularray#336]
Currently, if we try to use `\footnote` inside a `tblr` environment, the footnotes are going to go mental.

Sample code:
```tex
\documentclass{article}
\usepackage{tabularray}
\usepackage[language=english]{lipsum}

\title{Footnote Example}
\author{Mateus Felipe}
\date{December 2022}

\begin{document}

\maketitle

\section{Introduction}

\lipsum[1]\footnote{\lipsum[2]}

\begin{tblr}{c|c}
     Lorem\footnote{\lipsum[3]} & Ipsum  \\
     Dolor & Sit\footnote{\lipsum[4]}
\end{tblr}

\lipsum[5]\footnote{\lipsum[6]}

\end{document}
```

Result:
<img width="808" alt="image" src="https://user-images.githubusercontent.com/4605213/207466677-cf83827a-acf0-4c24-aa6b-bc0cb6d38dba.png">

This is actually a common problem with tabular environments in LaTeX. To solve this, I usually use `tabularx`, which simply works with footnotes. However, I prefer `tabularray` API way more. The only thing making me still use `tabularx` sometimes is this limitation.

**Comment by muzimuzhi** [comment]
Support for table footnotes (footnotes typeset right after table, not in normal page footnote area) is introduced in `tabularray` 2021L (see [changelog](https://github.com/lvjr/tabularray/wiki/ChangeLog#2021-07-01-version-2021l)), and used in several examples in package manual, but not documented yet.

Just search for `footnote` in the package manual and you'll see option `note` and command `\TblrNote` in examples.

**Comment by muzimuzhi** [comment]
Ah you want page footnotes. Then you may have to use  `\footnotemark` and `\footnotetext`, and manage `footnote` counter manually.
```tex
\documentclass{article}
\usepackage{tabularray}

\textheight=5cm % for screenshot

\begin{document}
Text\footnote{before tblr}

\begin{tblr}{colspec=lll, hlines, vlines}
  a\footnotemark[\numexpr\value{footnote}+1] & b \\
  c\footnotemark[\numexpr\value{footnote}+2] & d \\
\end{tblr}
\footnotetext[\numexpr\value{footnote}+1]{tabularray footnote}
\footnotetext[\numexpr\value{footnote}+2]{another tabularray footnote}
\addtocounter{footnote}{2}

Text\footnote{after tblr}
\end{document}
```
![image](https://user-images.githubusercontent.com/6376638/207671333-4dc155b8-84fa-46b8-b6d6-7f9f4552c7f5.png)

Automatic avoidance of counter jumping depends on the implementation of
- #179.

**Comment by mateusfccp** [comment]
> Ah you want page footnotes. Then you may have to use `\footnotemark` and `\footnotetext`, and manage `footnote` counter manually.

Yeah, I am aware of this way of making footnotes. The issue is precisely a request to be able to simply use the regular `\footnote` command inside it, like we can in `tabularx`. I am going to try to make the issue clearer about what it is about.

> Automatic avoidance of counter jumping depends on the implementation of

Thanks for the info, I am going to keep this one open as its related to it but not exactly the same thing.

**Comment by lvjr** [comment]
@mateusfccp You may have a look at the following answer of mine:

https://tex.stackexchange.com/questions/645421/tabularray-with-footnotes/645453#645453

**Comment by MrJake222** [comment]
Most annoying thing with the package. Please implement support for standard page `footnote`.
